rasterio/_warp.pyx

```diff
@@ -38,14 +40,19 @@ cdef extern from "gdalwarper.h" nogil:
                                 double dfProgressScale=1.0)
 
 
-RESAMPLING = namedtuple('RESAMPLING', [
-                'nearest', 
-                'bilinear', 
-                'cubic', 
-                'cubic_spline', 
-                'lanczos', 
-                'average', 
-                'mode'] )(*list(range(7)))
+class Resampling(IntEnum):
+    nearest=0
+    bilinear=1
+    cubic=2
+    cubic_spline=3
+    lanczos=4
+    average=5
+    mode=6
+    max=8
+    min=9
+    med=10
+    q1=11
+    q3=12
```

Comment:
@sgillies this is a breaking change for anyone using `RESAMPLING`.  For the short term, can you alias it and issue a deprecation warning for 1.0?